tests/test_find_cell/test_find_cell.py

```diff
@@ -0,0 +1,235 @@
+        small_clad_q1.setFill(self.materials['Clad'])
+        small_clad_q1.addSurface(halfspace=+1, surface=small_zcylinder)
+        small_clad_q1.addSurface(halfspace=-1, surface=medium_zcylinder)
+        small_clad_q1.addSurface(halfspace=+1, surface=xplane)
+        small_clad_q1.addSurface(halfspace=-1, surface=yplane)
+
+        small_moderator = openmoc.Cell(name='small pin moderator')
+        small_moderator.setNumSectors(8)
+        small_moderator.setFill(self.materials['Water'])
+        small_moderator.addSurface(halfspace=+1, surface=medium_zcylinder)
+
+        lattice_cell = openmoc.Cell(name='lattice cell')
+
+        root_cell = openmoc.Cell(name='root cell')
+        root_cell.addSurface(halfspace=+1, surface=boundaries[0])
+        root_cell.addSurface(halfspace=-1, surface=boundaries[1])
+        root_cell.addSurface(halfspace=+1, surface=boundaries[2])
+        root_cell.addSurface(halfspace=-1, surface=boundaries[3])
+        if (self.dimensions == 3):
+            root_cell.addSurface(halfspace=+1, surface=boundaries[4])
+            root_cell.addSurface(halfspace=-1, surface=boundaries[5])
+
+        pin1 = openmoc.Universe(name='large pin cell')
+        pin2 = openmoc.Universe(name='medium pin cell')
+        pin3 = openmoc.Universe(name='small pin cell')
+        assembly = openmoc.Universe(name='2x2 lattice')
+        root_universe = openmoc.Universe(name='root universe')
+
+        pin1.addCell(large_fuel)
+        pin1.addCell(large_moderator)
+        pin2.addCell(medium_fuel)
+        pin2.addCell(medium_moderator)
+        pin3.addCell(small_fuel)
+        pin3.addCell(small_clad_q1)
+        pin3.addCell(small_moderator)
+        assembly.addCell(lattice_cell)
+        root_universe.addCell(root_cell)
+
+        # 2x2 assembly
+        lattice = openmoc.Lattice(name='2x2 lattice')
+        lattice.setWidth(width_x=1.0, width_y=1.0)
+        lattice.setUniverses([[[pin1, pin2], [pin1, pin3]]])
+        lattice_cell.setFill(lattice)
+
+        # Rotate a lattice to test rotation of a fill cell
+        assembly_c = openmoc.Cell(name='rotated cell containing a lattice')
+        assembly_c.setFill(assembly)
+        assembly_c.setRotation((0,0,90), 'degrees')
+        assembly_r = openmoc.Universe(name='rotated lattice')
+        assembly_r.addCell(assembly_c)
+
+        # Translate a lattice to test translation of a fill cell
+        assembly_c = openmoc.Cell(name='rotated cell containing a lattice')
+        assembly_c.setFill(assembly)
+        assembly_c.setTranslation((0.05,0,0))
+        assembly_t = openmoc.Universe(name='translated lattice')
+        assembly_t.addCell(assembly_c)
+
+        # Rotate a lattice to test rotation of a fill cell
+        assembly_c = openmoc.Cell(name='translated cell containing a lattice')
+        assembly_c.setFill(assembly)
+        assembly_c.setRotation((0,0,-90), 'degrees')
+        #assembly_c.setTranslation((-1,0,0)) # doesn't seem to be supported
```

Comment:
I think I located the bug. They are lines 542-544 in Universe.cpp.
They should be 
          double x = next_coords->getX();
          double y = next_coords->getY();
          double z = next_coords->getZ();

Reply:
Good catch ! I saw these lines 1,000 times while doing this PR yet it didn't click